### PR TITLE
Check input is '...' in the trade page

### LINKED ISSUE #850: Fix: Decimal input is sometimes '...'
Sometimes the input for Decimal() is '...', causing a DecimalError.

### PR DESCRIPTION
**Pull request by tiagofilipenunes**
fix #850

### CODE REVIEW

File: src/components/trade/tradeWidget/useBuySell.ts [hunk 1]
```diff
@@ -173,7 +173,11 @@ export const useBuySell = ({
 
   useEffect(() => {
     if (byTargetQuery.data) {
-      if (new Decimal(targetInput).gt(new Decimal(liquidityQuery.data || 0))) {
+      if (
```
Comment (GrandSchtroumpf) [inline src/components/trade/tradeWidget/useBuySell.ts:176]: It's not super readable, could you create variables with good names so it's easier to understand what is this condition about

File: src/components/trade/tradeWidget/useTradeAction.ts [hunk 2]
```diff
@@ -49,8 +49,12 @@ export const useTradeAction = ({
 
   const calcMaxInput = useCallback(
     (amount: string) => {
+      if (amount === '...') return 'NaN';
```
Comment (GrandSchtroumpf) [inline src/components/trade/tradeWidget/useTradeAction.ts:52]: Why do you return NaN as a string ? It doesn't look like a good practice

File: src/components/trade/tradeWidget/useBuySell.ts (hunk 1 above)
Comment (tiagofilipenunes) [reply src/components/trade/tradeWidget/useBuySell.ts:177]: Made it clearer

Comment (zavelevsky) [reply src/components/trade/tradeWidget/useBuySell.ts:177]: we can consider creating a utility function, something in the lines of (coding from memory)
```ts

function tryDecimal(value: number|string|Decimal) : Decimal | NaN {
  try {
    return new Decimal(value);
  } catch {
    return NaN;
  }
}

```

Comment (zavelevsky) [reply src/components/trade/tradeWidget/useBuySell.ts:177]: and of course use it only when the input is questionable

Comment (tiagofilipenunes) [reply src/components/trade/tradeWidget/useBuySell.ts]: Updated the implementation

**Pull request merged**